IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -38,11 +38,28 @@ + (NSString *_Nullable)getWPJStringDataForIdentifier:(nonnull NSString *)identif
                                          accessGroup:(nullable NSString *)accessGroup
                                              context:(id<MSIDRequestContext>_Nullable)context
                                                error:(NSError*__nullable*__nullable)error
+{
+    return [self getWPJStringDataFromV2ForTenantId:nil
+                                        identifier:identifier
+                                       accessGroup:accessGroup
+                                           context:context
+                                             error:error];
+}
+
++ (NSString *_Nullable)getWPJStringDataFromV2ForTenantId: (NSString *)tenantId
+                                              identifier:(nonnull id)identifier
+                                             accessGroup:(nullable NSString *)accessGroup
+                                                 context:(id<MSIDRequestContext>_Nullable)context
+                                                   error:(NSError*__nullable*__nullable)error
 {
     // Building dictionary to retrieve given identifier from the keychain
     NSMutableDictionary *query = [[NSMutableDictionary alloc] init];
     [query setObject:(__bridge id)(kSecClassGenericPassword) forKey:(__bridge id<NSCopying>)(kSecClass)];
-    [query setObject:identifier forKey:(__bridge id<NSCopying>)(kSecAttrAccount)];
+    if (tenantId) {
```

Comment:
Given that first registration and Intune associated registration is always written to v1 keychain, 
- If user DOES NOT pass tenantId to getWPJMetadata, it will find the entry in v1 and would return the details as existing MSAL already does
- If user passes tenantId to getWPJMetadata, it will find the entry in v1 and would hit: 
**getWPJStringDataForIdentifier** instead of **getWPJStringDataFromV2ForTenantId**. **getWPJStringDataForIdentifier** passes nil as tenantId to **getWPJStringDataFromV2ForTenantId**. So, code flow would be the same as that of existing MSAL.

Please let me know if I've missed some case.